Initial integration of websockets inside Kura.

MMaiero opened the pull request:
Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero committed:
Initial integration of websockets inside Kura.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

kartben commented on kura/org.eclipse.kura.core/OSGI-INF/metatype/org.eclipse.kura.core.data.transport.mqtt.MqttDataTransport.xml:
```diff
@@ -26,7 +26,7 @@
             required="true"
             default="mqtt://broker-url:1883/" 
-            description="URL of the mqtt broker to connect to, for example, mqtt://broker-sandbox.everyware-cloud.com:1883/ or mqtts://broker-sandbox.everyware-cloud.com:8883/."/>
+            description="URL of the mqtt broker to connect to, for example, mqtt://broker-sandbox.everyware-cloud.com:1883/, mqtts://broker-sandbox.everyware-cloud.com:8883/, ws://broker-sandbox.everyware-cloud.com:8080/ or wss://broker-sandbox.everyware-cloud.com:443/."/>
```
please include iot.eclipse.org sandbox broker in this list. Listing only everyware-cloud doesn't look right to me :-)

ctron replied:
I agree!

MMaiero committed:
Updated help message in metatype, changed code in disconnect.

The disconnect code was updated adding a delay, in order to overcome a sincronization issue
with the websockets code.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero committed:
Minor clenups, reorderings and optimizations.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

MMaiero commented:
Closes #556
